packages/twenty-front/src/modules/analytics/utils/fetchGraphDataOrThrow.ts

```diff
@@ -0,0 +1,40 @@
+import { ANALYTICS_GRAPH_OPTION_MAP } from '@/analytics/constants/AnalyticsGraphOptionMap';
+import { computeStartEndDate } from '@/analytics/utils/computeStartEndDate';
+type fetchGraphDataOrThrowProps = {
+  recordId: string;
+  recordType: string;
+  windowLength: '7D' | '1D' | '12H' | '4H';
+  tinybirdJwt: string;
+  endpointName: string;
+};
+
+export const fetchGraphDataOrThrow = async ({
+  recordId,
+  recordType,
+  windowLength,
+  tinybirdJwt,
+  endpointName,
+}: fetchGraphDataOrThrowProps) => {
+  const queryString = new URLSearchParams({
+    ...ANALYTICS_GRAPH_OPTION_MAP[windowLength],
+    ...computeStartEndDate(windowLength),
+    ...(recordType === 'webhook'
+      ? { webhookId: recordId }
+      : { functionId: recordId }), //later create a switch case
```

Comment:
concatenate recordType + Id (recordType which you should get from a constant map, cf other comment)